docs/enhancements/agent-late-binding.md

```diff
@@ -0,0 +1,126 @@
+
+### Goals
+Note: The language here refers to CRDs, but also holds true for the corresponding REST resources
+
+- Make the InfraEnv and Agent resources independent from the ClusterDeployment. A single InfraEnv should have the potential to boot any number of hosts, whose Agents register with no relation to any particular ClusterDeployment.  An Agent may be assigned to a ClusterDeployment at a later time.
+
+- An Agent may be associated with a different ClusterDeployment at any time, unless it is part of an ongoing cluster installation.
+
+### Non-Goals
+Note: The language here refers to CRDs, but also holds true for the corresponding REST resources
+
+- Once a host boots from disk, booting the Discovery ISO again to return it to the collection of free Agents is considered to be a manual step.  There are several options that we will explore whereby this can be done automatically, but that discussion is out of scope for this enhancement.
+
+- At a later stage, Agents may be automatically assigned to clusters based on their discovered inventory and/or network configurations. This is future work and not in the scope of this enhancement.
+
+- For SaaS, there are discussions about changing the tenancy model to include organizations and not only users.  Those decisions are out of scope for this document.
+
+## Proposal
+
+There are two main (non-breaking) changes to the CRDs:
+1. ClusterRef in the InfraEnv Spec will no longer be mandatory.
+1. ClusterRef in the Agent Spec will no longer be mandatory.
+
+A Discovery ISO with no associated cluster means that hosts booted with that ISO will have agents that are not bound to any cluster.  Thus, Agent CRs can be created with no cluster binding.  It will still be possible to retain the previous behavior by providing a ClusterRef.  The tenancy model will remain unchanged, with Agents being created in the namespace of the InfraEnv they booted from.
+
+The Agent's ClusterRef may now be deleted or changed dynamically, unless the Agent is part of an ongoing cluster installation, in which case the installation must first be cancelled (not currently possible via the k8s API).  If the host had started/completed its installation, a new Agent Condition would indicate that the user needs to boot the host into the Discovery ISO once
+again (as mentioned in the non-goals, this can be done automatically but is not in the scope of this enhancement).
+
+The REST API will require breaking changes, so we will create a v2 API.  At the same time we will align with the k8s API (change host to agent, and image to infra-env).
+1. Move images (infra-envs) to be a standalone resource, rather than part of the cluster resource (currently `/clusters/{cluster_id}/downloads/image`).  This removes the dependency of an image on a specific cluster, removes the current limitation of one image per cluster, and aligns with InfraEnv CRD.
```

Comment:
Is it safe to assume the API for the image would be something like `/infra-envs/{infra_env_id}/downloads/image`?
Also, would the structure for an infra-env resource in the REST API try to mirror the infra-env kube api?

Reply:
make sense to me

Reply:
I updated the enhancement to clarify that we are moving `/clusters/{cluster_id}/downloads/image` to `/infra-envs`.  So instead of metadata for a single image embedded in a cluster resource, it has the full CRUD (and download ISO API).